src/lightning/pytorch/loggers/comet.py

```diff
@@ -271,50 +261,37 @@ def experiment(self) -> Union["Experiment", "ExistingExperiment", "OfflineExperi
         if self._experiment is not None and self._experiment.alive:
             return self._experiment
 
-        if self._future_experiment_key is not None:
-            os.environ["COMET_EXPERIMENT_KEY"] = self._future_experiment_key
-
-        from comet_ml import ExistingExperiment, Experiment, OfflineExperiment
-
-        try:
-            if self.mode == "online":
-                if self._experiment_key is None:
-                    self._experiment = Experiment(api_key=self.api_key, project_name=self._project_name, **self._kwargs)
-                    self._experiment_key = self._experiment.get_key()
-                else:
-                    self._experiment = ExistingExperiment(
-                        api_key=self.api_key,
-                        project_name=self._project_name,
-                        previous_experiment=self._experiment_key,
-                        **self._kwargs,
-                    )
-            else:
-                self._experiment = OfflineExperiment(
-                    offline_directory=self.save_dir, project_name=self._project_name, **self._kwargs
-                )
-            self._experiment.log_other("Created from", "pytorch-lightning")
-        finally:
-            if self._future_experiment_key is not None:
-                os.environ.pop("COMET_EXPERIMENT_KEY")
-                self._future_experiment_key = None
-
-        if self._experiment_name:
-            self._experiment.set_name(self._experiment_name)
+        import comet_ml
+
+        comet_comfig = comet_ml.ExperimentConfig(
+            offline_directory=self._save_dir,
+            name=self._experiment_name,
+            **self._kwargs,
+        )
+
+        self._experiment = comet_ml.start(
+            api_key=self.api_key,
+            project=self._project_name,
+            experiment_key=self._experiment_key,
+            online=self.mode == "online",
+            experiment_config=comet_comfig,
+        )
+
+        self._experiment.log_other("Created from", "pytorch-lightning")
 
         return self._experiment
 
     @override
     @rank_zero_only
     def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
         params = _convert_params(params)
-        params = _flatten_dict(params)
```

Comment:
It's good that we don't flatten the dict anymore, did you see a difference in the reported parameters when running the examples?

Reply:
for logged parameter:
comet_logger.log_hyperparams({"specific": {'param': {'subparam': "value"}}})

it will look:

name | value
specific | {"param": {"subparam": "value"}}